File: src/main/java/fr/lewon/web/bot/entities/battle/AbstractBattleOpponent.java
package fr.lewon.web.bot.entities.battle;

import java.math.BigDecimal;

import com.fasterxml.jackson.annotation.JsonProperty;

public class AbstractBattleOpponent {

	@JsonProperty
	private Integer orgasm;

	@JsonProperty
	private BigDecimal ego;

	@JsonProperty
	private Integer x;

	@JsonProperty("curr_ego")
	private BigDecimal currEgo;

	@JsonProperty("nb_org")
	private Integer nbOrg;

	@JsonProperty
	private Integer figure;

	public Integer getOrgasm() {
		return orgasm;
	}

	public void setOrgasm(Integer orgasm) {
		this.orgasm = orgasm;
	}

	public BigDecimal getEgo() {
		return ego;
	}

	public void setEgo(BigDecimal ego) {
		this.ego = ego;
	}

	public Integer getX() {
		return x;
	}

	public void setX(Integer x) {
		this.x = x;
	}

	public BigDecimal getCurrEgo() {
		return currEgo;
	}

	public void setCurrEgo(BigDecimal currEgo) {
		this.currEgo = currEgo;
	}

	public Integer getNbOrg() {
		return nbOrg;
	}

	public void setNbOrg(Integer nbOrg) {
		this.nbOrg = nbOrg;
	}

	public Integer getFigure() {
		return figure;
	}

	public void setFigure(Integer figure) {
		this.figure = figure;
	}

}

File: src/main/kotlin/fr/lewon/bot/hh/entities/battle/AbstractBattleOpponent.kt
package fr.lewon.bot.hh.entities.battle

import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal

open class AbstractBattleOpponent {
    @JsonProperty
    var orgasm: Int? = null
    @JsonProperty
    var ego: BigDecimal? = null
    @JsonProperty
    var x: Int? = null
    @JsonProperty("curr_ego")
    var currEgo: BigDecimal? = null
    @JsonProperty("nb_org")
    var nbOrg: Int? = null
    @JsonProperty
    var figure: Int? = null

}